pina/utils.py

```diff
@@ -25,12 +26,21 @@ def check_consistency(object, object_instance, object_name, subclass=False):
     :raises ValueError: If the object does not inherit from the
         specified class
     """
-    if not subclass:
-        if not isinstance(object, object_instance):
-            raise ValueError(f"{object_name} must be {object_instance}")
+    def check_inheritance(obj):
```

Comment:
You can also avoid to define a function inside a function. You need to preprocess first object. If is an object, make a list with just a single element. Otherwise use a list. So you can remove  line 39 to 43